src/runtime_src/core/tools/xbmgmt2/flash/firmware_image.cpp

```diff
@@ -373,8 +374,7 @@ bool DSAInfo::matchIntId(std::string &id) const
 
```

Comment:
There is A LOT going on in this small method (e.g., member variables comparison, removal of hex "0x" leading characters, assumptions being made with strings, etc.)

Please rewrite this method to be more readable and add comment on what is being done.